src/proof/nova.rs

```diff
@@ -241,110 +241,104 @@ impl<F: LurkField> Nova<F> for NovaProver<F> {
+        expr: &str,
+        expected_result: Option<Ptr<Fr>>,
+        expected_env: Option<Ptr<Fr>>,
+        expected_cont: Option<ContPtr<Fr>>,
+        expected_emitted: Option<Vec<Ptr<Fr>>>,
+        expected_iterations: usize,
+        chunk_frame_count: usize,
+    ) {
+        let limit = 100000;
+        let expr = s.read(expr).unwrap();
 
         let e = empty_sym_env(&s);
 
         let nova_prover = NovaProver::<Fr>::new(chunk_frame_count);
-        let proof_results = if check_nova {
-            Some(
-                nova_prover
-                    .evaluate_and_prove(expr, empty_sym_env(&s), &mut s, limit)
-                    .unwrap(),
-            )
-        } else {
-            None
-        };
+        let proof_results = Some(
+            nova_prover
+                .evaluate_and_prove(expr, empty_sym_env(&s), s, limit)
+                .unwrap(),
+        );
 
         let shape_and_gens = nova_prover.make_shape_and_gens();
 
-        if check_nova {
-            if let Some((proof, instance)) = proof_results {
-                proof.verify(&shape_and_gens, &instance);
-            }
+        if let Some((proof, instance)) = proof_results {
+            proof.verify(&shape_and_gens, &instance);
         }
 
-        if check_constraint_systems {
-            let frames = nova_prover.get_evaluation_frames(expr, e, &mut s, limit);
-
-            let multiframes = MultiFrame::from_frames(nova_prover.chunk_frame_count(), &frames, &s);
-            let cs = nova_prover.outer_synthesize(&multiframes).unwrap();
+        let frames = nova_prover.get_evaluation_frames(expr, e, s, limit);
 
-            let adjusted_iterations = nova_prover.expected_total_iterations(expected_iterations);
-            let output = cs[cs.len() - 1].0.output.unwrap();
+        let multiframes = MultiFrame::from_frames(nova_prover.chunk_frame_count(), &frames, &s);
+        let cs = nova_prover.outer_synthesize(&multiframes).unwrap();
 
-            if !debug {
-                dbg!(
-                    multiframes.len(),
-                    nova_prover.chunk_frame_count(),
-                    frames.len(),
-                    output.expr.fmt_to_string(&s)
-                );
+        let adjusted_iterations = nova_prover.expected_total_iterations(expected_iterations);
+        let output = cs[cs.len() - 1].0.output.unwrap();
 
-                assert_eq!(expected_iterations, Frame::significant_frame_count(&frames));
-                assert_eq!(adjusted_iterations, cs.len());
-                let status: Status = output.cont.into();
-                assert_eq!(expected_cont, status);
-                if expected_cont != Status::Error {
-                    assert_eq!(expected_result, output.expr);
-                }
-            }
+        // dbg!(
+        //     multiframes.len(),
+        //     nova_prover.chunk_frame_count(),
+        //     frames.len(),
+        //     output.expr.fmt_to_string(&s)
+        // );
```

Comment:
You might as well leave these uncommented, as useful if the test fails (my opinion — and why they were there).

Reply:
Done.